### ISSUE TITLE
KeyError when saving Image as GIF format

### ISSUE THREAD

**Issue by Maslino** [python-pillow/Pillow#831]
Hello, I encounter a KeyError with Pillow 2.5.1 when saving Image as GIF format.  I tried Pillow 2.5.1, 2.5.0, 2.4.0 and 2.1.0 and finally found that only 2.1.0 worked well.

## Environment
CentOS 6.4 64bit, Python 2.7.8, Pillow 2.5.1, qrcode 5.0.1(for generating QR Code)

## PIL setup summary
    version      Pillow 2.5.1
    platform     linux2 2.7.8 (default, Jul 31 2014, 11:37:40)
                 [GCC 4.4.7 20120313 (Red Hat 4.4.7-3)]
    --------------------------------------------------------------------
    --- TKINTER support available
    --- JPEG support available
    *** OPENJPEG (JPEG2000) support not available
    --- ZLIB (PNG/ZIP) support available
    *** LIBTIFF support not available
    --- FREETYPE2 support available
    --- LITTLECMS2 support available
    *** WEBP support not available
    *** WEBPMUX support not available
    --------------------------------------------------------------------
## How to repeat
```python
from qrcode import QRCode
from StringIO import StringIO
qrc = QRCode()
qrc.add_data("hello world")
image = qrc.make_image()
image.save(StringIO(), "GIF")
```
## Exception traceback
```python
Traceback (most recent call last):
  File "<ipython-input-8-de5754d11441>", line 2, in <module>
    image.save(StringIO(), "GIF")
  File "/home/liuxiong/local/python/lib/python2.7/site-packages/qrcode/image/pil.py", line 32, in save
    self._img.save(stream, kind)
  File "/home/liuxiong/local/python/lib/python2.7/site-packages/PIL/Image.py", line 1663, in save
    save_handler(self, fp, filename)
  File "/home/liuxiong/local/python/lib/python2.7/site-packages/PIL/GifImagePlugin.py", line 262, in _save
    header, usedPaletteColors = getheader(imOut, palette, im.encoderinfo)
  File "/home/liuxiong/local/python/lib/python2.7/site-packages/PIL/GifImagePlugin.py", line 427, in getheader
    imageBytes[i] = newPositions[imageBytes[i]]
KeyError: 192
```

**Comment by aclark4life** [comment]
@wiredfool What does your snippage demonstrate? Is this a Pillow bug?

**Comment by radarhere** [comment]
I tried to replicate this error, and wasn't able to. Looking into it, I figured out why. 

Merged pull request #993 means that optimization only applies to P or L mode images. Since this is a 1 mode image, the line in GifImagePlugin.py that produces the error isn't run anymore.

**Comment by hugovk** [comment]
Closing.

Cannot reproduce on OSX / Python 2.7.9  / Pillow 2.8.1.

Thanks for checking @radarhere.
